### PR TITLE
[9.0] Migrate mail_optional_follower_notification

### PR DESCRIPTION
**Pull request by lmignon**

### CODE REVIEW

**Comment by hbrunn** [comment]
please get the commits from 8.0 to allow simple cherry picking of further developments in 8.0: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#howto

**Comment by rafaelbn** [comment]
Waiting for the fix in order to test it. Thanks

**Comment by lmignon** [comment]
@hbrunn This branch is based on the one merged in 8.0. It seems that the original commits have been squashed when the branch has been merged. Why? I've asked @adrienpeiffer if he is aware of this squash and he isn't....  Now I've to redo my branch and I'm not happy with that.

**Comment by hbrunn** [comment]
@lmignon looking at the commit log, it seems I'm the culprit. Then ignore my request, and I'll look into how to avoid those mistakes in the future. Apologies for the noise.

**Comment by hbrunn** [comment]
everything else looks good btw, :+1:

**Comment by lmignon** [comment]
@hbrunn Rebase done...

**Comment by lmignon** [comment]
@hbrunn 

> @lmignon looking at the commit log, it seems I'm the culprit. Then ignore my request, and I'll look into how to avoid those mistakes in the future. Apologies for the noise.

I don't blame you, In this case, the 'rebase' was easy. (reset --hard oca/9.0 + cherry-pick commits on 8.0 + cherry-pick specific commits on 9.0 = 5 commands :smirk: )

Unfortunately, it seems that I've to fix my branch...

**Comment by lmignon** [comment]
@hbrunn Can yout put the label WIP? I've to fix the tests but didn't have time now.

**Comment by rafaelbn** [comment]
Hi @lmignon , which is the **difference** between this module and configure the same in Settings - Email - Subtypes?

I know that this is a migration, but in fact module in v8 was not necessary as you can configure it.

**Comment by adrienpeiffer** [comment]
@rafaelbn What do you mean about `configure the same` ?
 We can use this module without a subtype. it's not the same feature. Here, the user can simply decide if he want to notify follower or not without any configuration.

File: mail_optional_follower_notification/models/res_partner.py
Comment (adrienpeiffer) [inline mail_optional_follower_notification/models/res_partner.py]: Don't propagate default value of named arguments

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]

**Comment by dreispt** [comment]
I did not read the code, but from the module description, this seems to be the same feature as the standard "Log a message" - it writes a message on the Chatter without sending it to the Followers.

**Comment by adrienpeiffer** [comment]
@dreispt You have the possibility to define some recipent but you can exclude followers.

**Review by dreispt — APPROVED** [review]
I suggest a better description:
"When sending a message in a document's mail chatter, all Followers are notified.
This feature allows to uncheck the notify followers checkbox, and have the notification sent only to selected people.
Beware that the message will still be visible to all followers when viewing the document."

But not blocking and implementation LGTM

**Comment by lmignon** [comment]
rebased

File: mail_optional_follower_notification/tests/test_mail_optional_follower_notifications.py
Comment (hbrunn) [inline mail_optional_follower_notification/tests/test_mail_optional_follower_notifications.py]: this should be `needaction_partner_ids` by now: https://github.com/OCA/OCB/blob/9.0/addons/mail/models/mail_message.py#L950

Comment (hbrunn) [reply mail_optional_follower_notification/tests/test_mail_optional_follower_notifications.py]: @lmignon can you fix the test? Then we can merge I think

**Comment by lmignon** [comment]
@hbrunn I've edited the file on github If tests are OK I'll squash my last commits

**Comment by hbrunn** [comment]
yay, worked. I'll count @adrienpeiffer as a +1 because his comment was addressed too, so I'll merge after the squashing

**Comment by lmignon** [comment]
@hbrunn Commit squahed...

**Pull request merged**
